### PR TITLE
Revert "Non-allocating isinplace"

### PR DESCRIPTION
**Pull request by kanav99**
Reverts SciML/DiffEqBase.jl#601

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "DiffEqBase"
 uuid = "2b5f629d-d688-5b77-993f-72d75c75574e"
 authors = ["Chris Rackauckas <accounts@chrisrackauckas.com>"]
-version = "6.48.3"
+version = "6.48.2"
```
Comment (kanav99) [inline Project.toml:4]: We need to keep it 6.48.3 right?

Comment (ChrisRackauckas) [reply Project.toml:4]: yes, we also need to setup Github actions.

Comment (kanav99) [reply Project.toml:4]: yeah let me do that in this PR

File: .github/workflows/Test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - master
+    tags: '*'
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (devmotion) [inline .github/workflows/Test.yml:12]: Maybe better to run two tests in parallel?
```suggestion
    runs-on: ubuntu-latest
    strategy:
      matrix:
        group:
          - Core
          - Downstream
```

File: .github/workflows/Test.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        run: julia --project=./ -e 'using Pkg; Pkg.instantiate()'
+      - name: Test
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # For authentication with GitHub Actions token
+        run: julia --project=./ -e 'using Pkg; pkg"test";'
+      - name: Test
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # For authentication with GitHub Actions token
+          GROUP: Downstream
+        run: julia --project=./ -e 'using Pkg; pkg"test";'
```
Comment (devmotion) [inline .github/workflows/Test.yml:28]: Just use the default actions and cache artifacts?
```suggestion
      - uses: actions/cache@v1
        env:
          cache-name: cache-artifacts
        with:
          path: ~/.julia/artifacts
          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
          restore-keys: |
            ${{ runner.os }}-test-${{ env.cache-name }}-
            ${{ runner.os }}-test-
            ${{ runner.os }}-
      - uses: julia-actions/julia-buildpkg@latest
      - uses: julia-actions/julia-runtest@latest
        env:
          GROUP: ${{ matrix.group }}
```

**Review by devmotion — COMMENTED** [review]
I added some suggestions. It would also be good to add some coverage analysis, such as e.g. in https://github.com/TuringLang/DynamicPPL.jl/blob/cf95183bb085eaacaea69d7c91dfc8c9c3b45f17/.github/workflows/DynamicPPL-CI.yml#L58-L68.

**Comment by kanav99** [comment]
I have applied your suggestions, let us see if there is any issue in that.

**Pull request merged**